File: 2971.py
class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        prefixSum = sum(nums)
        for x in nums[:-2]:
            if x * 2 < prefixSum:
                return prefixSum
            prefixSum -= x
        return -1

File: sol/2971/2971.cpp
struct Solution {
    long long largestPerimeter(vector<int>& nums) {
        sort(nums.begin(), nums.end(), std::greater{});
        long long prefix = 0;
        for (int &x : nums){
            prefix += x;
        }
        for (int i = 0; i < nums.size() - 2; ++i){
            if ((nums[i] << 1) < prefix){
                return prefix;
            }
            prefix -= nums[i];
        }
        return -1;
    }
};
